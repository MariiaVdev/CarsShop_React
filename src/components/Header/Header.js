import React from 'react';
import styles from './Header.module.scss';
import PropTypes from 'prop-types';

class Header extends React.PureComponent {

    render() {
        const { title, isFav, inCart } = this.props;
        return (
            <header>
                <span className={styles.headerTitle}>{title}</span>
                <div className={styles.headerContainer}>
                    <img src='./img/favorite_Not.png' alt='Favorite' />
                    <span className={styles.headerCount}>{isFav}</span>
                    <img className={styles.headerCart} src='./img/shopping-cart.png' alt='Cart' />
                    <span className={styles.headerCount}>{inCart}</span>
                </div>
            </header>)

    }
}

Header.propTypes = {
    title: PropTypes.string,
    isFav: PropTypes.number,
    inCart: PropTypes.number


}

Header.defaultProps = {
    title: 'Lexus',
    isFav: 0,
    inCart: 0
}

export default Header;