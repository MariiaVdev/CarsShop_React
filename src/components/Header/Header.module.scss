header {
    display: flex;
    align-items: center;
    box-sizing: border-box;
    padding: 0 36px;
    background: #4d4d52;
    color: #fff;
    width: 100%;
    height: 60px;
    position: fixed;
    top: 0;
    left: 0; 
    
    a {
      text-decoration: none;
      color:  #fff;
    }
  }
  
  .headerTitle {
    font-size: 26px;
    text-transform: uppercase;
  }
  
  .headerContainer {
    display: flex;
    align-items: center;
    margin-left: 60vw;
  
    img {
      width: 25px;
      height: 25px;
      margin-right: 5px;
      margin-top: 5px;
    }
    .headerCart {
        width: 27px;
        height: 27px;
        margin-top: 7px;
      }
    .headerCount {
        font-size: 20px;
        margin-right: 20px;
    }
  }
  