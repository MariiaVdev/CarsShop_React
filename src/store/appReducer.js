import {combineReducers} from "redux";
import modalReducer from "./modal/modalReducer";
import carsReducer from './cars/carsReducer';
import cardReducer from './card/cardReducer';
import cartReducer from './cart/cartReducer';
import formReducer from "./form/formReducer";





const appReducer = combineReducers({
    modal: modalReducer,
    cars: carsReducer,
    card: cardReducer,
    cart: cartReducer,
    form: formReducer
})


export default appReducer;