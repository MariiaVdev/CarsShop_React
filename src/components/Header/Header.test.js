import { render } from "@testing-library/react"
import Header from './Header'
import { Provider } from "react-redux"
import store from "../../store"
import { BrowserRouter } from "react-router-dom"

const Wrapper = () => {
    return (
        <>
        <BrowserRouter>
        <Provider store={store}>
        <Header />
        </Provider>
        </BrowserRouter>
        </>
    )

}
describe("Render Header", () => {
    test("should Header render with default props", () => {
        const { asFragment } = render(<Wrapper/>);
        expect(asFragment()).toMatchSnapshot()
    })
})


