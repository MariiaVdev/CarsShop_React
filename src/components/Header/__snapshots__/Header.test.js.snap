// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header should Header render with default props 1`] = `
<DocumentFragment>
  <header>
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      <span
        class="headerTitle"
      >
        Lexus
      </span>
    </a>
    <div
      class="headerContainer"
    >
      <a
        class=""
        href="/favorite"
      >
        <img
          alt="Favorite"
          src="./img/favorite_Not.png"
        />
      </a>
      <span
        class="headerCount"
      >
        0
      </span>
      <a
        class=""
        href="/carts"
      >
        <img
          alt="Cart"
          class="headerCart"
          src="./img/shopping-cart.png"
        />
      </a>
      <span
        class="headerCount"
      >
        0
      </span>
    </div>
  </header>
</DocumentFragment>
`;
